specifyweb/frontend/js_src/lib/components/Statistics/StatsSpec.tsx

```diff
@@ -0,0 +1,502 @@
+import { getDateInputValue } from '../../utils/dayJs';
+import type { IR } from '../../utils/types';
+import { formatNumber } from '../Atoms/Internationalization';
+import { queryFieldFilters } from '../QueryBuilder/FieldFilter';
+import type { BackendStatsResult, BackendStat, StatItemSpec } from './utils';
+import { statsText } from '../../localization/stats';
+import { formattedEntry } from '../WbPlanView/mappingHelpers';
+
+function modifyBackendResult<CATEGORY_NAME extends keyof BackendStatsResult>(
+  backendobject: BackendStatsResult[CATEGORY_NAME] | undefined,
+  modifyfunction: (
+    rawValue: BackendStatsResult[CATEGORY_NAME][keyof BackendStatsResult[CATEGORY_NAME]]
+  ) => BackendStat['value']
+): StatCategoryReturn {
+  if (backendobject === undefined) {
+    return undefined;
+  }
+  return Object.fromEntries(
+    Object.entries(backendobject).map(([key, value]) => [
+      key,
+      {
+        label: key,
+        spec: {
+          type: 'Backendstat',
+          value: modifyfunction(value),
+        },
+      },
+    ])
+  );
+}
+
+export type StatCategoryReturn =
+  | IR<{ readonly label: string; readonly spec: StatItemSpec }>
+  | undefined;
+
+type StatsSpec =
+  | {
+      readonly [CATEGORY_NAME in keyof BackendStatsResult]: {
+        readonly label: string;
+        readonly categories: (
+          backendStatsResult: BackendStatsResult[CATEGORY_NAME] | undefined
+        ) => StatCategoryReturn;
+      };
+    }
+  | {
+      readonly [CATEGORY_NAME in string]: {
+        readonly label: string;
+        readonly categories: () => StatCategoryReturn;
+      };
+    };
+
+export const statsSpec: IR<StatsSpec> = {
+  collection: {
+    holdings: {
+      label: statsText('holdings'),
+      categories(backendStatsResult) {
+        return {
```

Comment:
[not blocking] could you get rid of return here in favor if implicit return as there is just one expression in the function? same for other places in this file